manticore/core/manticore.py

```diff
@@ -373,6 +361,11 @@ def __init__(self, initial_state, workspace_url=None, outputspace_url=None, **kw
 
         # Workers will use manticore __dict__ So lets spawn them last
         self._workers = [self._worker_type(id=i, manticore=self) for i in range(consts.procs)]
+
+        # We won't create the daemons until .run() is called
+        self._daemon_threads = []
```

Comment:
Frequently, when running Manticore, I find myself calling `run` in one thread, and having another thread doing something else - like intermittently printing instructions/second or communicating with another process. The daemon_threads API formalizes that and makes it easy to incorporate into Manticore without manually constructing an additional thread. 

In this PR, daemon threads are only used for testing. However, I ripped them out of the TUI PR, which uses them to run the socket server that the TUI talks to.

Reply:
ah! here it is that reason.
So it is a convenient thing. Maybe instead of registering an keeping "_daemon_threads" "_daemon_callbacks" 

```
m = Manticore()
with m.Daemon(callback=somefunction):
    m.run()
```
or 
```
m = Manticore()
m.run(daemon_callback=somefunction)
```